Restructure and consolidate information in Contribute section (#19576)
kubernetes/website: content/en/docs/contribute/new-content/contributing-content.md

---
title: Opening a pull request

> slug: new-content

Reviewer: nit: This file, `Opening a pull request`, is reached by `/docs/contribute/new-content/new-content/`. These different names/phrases will make it harder to understand/associate what is in the page.

Author: The next commit will rename this to `open-a-pr.md`.

1. In your local repository, check which files you need to commit:

    ```bash
    git status
    ```

    Output is similar to:

    ```bash
    On branch <my_new_branch>
    Your branch is up to date with 'origin/<my_new_branch>'.

    Changes not staged for commit:
    (use "git add <file>..." to update what will be committed)
    (use "git checkout -- <file>..." to discard changes in working directory)

    modified:   content/en/docs/contribute/new-content/contributing-content.md

    no changes added to commit (use "git add" and/or "git commit -a")
    ```

2. Add the files listed under **Changes not staged for commit** to the commit:

    ```bash
    git add <your_file_name>
    ```

    Repeat this for each file.

3.  After adding all the files, create a commit:

    ```bash
    git commit -m "Your commit message"
    ```

    {{< note >}}
    Do not reference a GitHub issue or pull request by ID or URL in the
    commit message. You can link issues and pull requests together later in the GitHub UI.
    {{< /note >}}

4. Push your local branch and its new commit to your remote fork:

    ```bash
    git push origin <my_new_branch>
    ```

> Next:

Reviewer: nit: Since the rest of the Contribute sections where streamlined, these new content additions (fork+clone, rebasing, squashing) could be pared down.
nit: The `Next` and the two items after don't seem to fit in when you have sub-headings immediated after.

---
title: Opening a pull request
slug: new-content
content_template: templates/concept
weight: 10
card:
  name: contribute
  weight: 40
---

{{% capture overview %}}

{{< note >}}
**Code developers**: If you are documenting a new feature for an
upcoming Kubernetes release, see
[Document a new feature](/docs/contribute/new-content/new-features/).
{{< /note >}}

To contribute new content pages or improve existing content pages, open a pull request (PR). Make sure you follow all the requirements in the [Before you begin](#before-you-begin) section.

If your change is small, or you're unfamiliar with git, read [Changes using GitHub](#changes-using-github) to learn how to edit a page.

> If your changes are large, read [Work from a local fork](#fork-the-repo) to learn how to make changes on your locally, on your computer.

Reviewer 2: ```suggestion
If your changes are large, read [Work from a local fork](#fork-the-repo) to learn how to make changes locally on your computer.
```

{{% /capture %}}

{{% capture body %}}

## Changes using GitHub

> If you're less experienced with git workflows, this is an easier method of

Reviewer 2: ```suggestion
If you're less experienced with git workflows, here's an easier method of
```

opening a pull request.

1.  On the page where you see the issue, select the pencil icon at the top right.

>     You can also scroll to the bottom of the page and select **Edit this page**.

Author: Re: Select -> This is as per [Microsoft's Style Guide](https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui).

2.  Make your changes in the GitHub markdown editor.

3.  Below the editor, fill in the **Propose file change**
    form. In the first field, give your commit message a title. In
    the second field, provide a description.

    {{< note >}}
    Do not use any [GitHub Keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) in your commit message. You can add those to the pull request
    description later.

>     {{< /note >}}

Author: I had a long think on this – the only more visible/central place put this would be in the page's Overview, given the current IA. 

I don't think putting it as a note/warning at the top of the task makes sense either – it's applicable directly to the step about writing commit messages. 

Finally – the "advanced" open a PR topic also has a note along the same lines, but for good measure I changed the wording so it's the same as this one.

It's not ideal, but I don't think there's a better solution.

{{% capture overview %}}

{{< note >}}
**Code developers**: If you are documenting a new feature for an
upcoming Kubernetes release, see
[Document a new feature](/docs/contribute/new-content/new-features/).
{{< /note >}}

To contribute new content pages or improve existing content pages, open a pull request (PR). Make sure you follow all the requirements in the [Before you begin](#before-you-begin) section.

If your change is small, or you're unfamiliar with git, read [Changes using GitHub](#changes-using-github) to learn how to edit a page.

> If your changes are large, read [Work from a local fork](#fork-the-repo) to learn how to make changes on your locally, on your computer.

Reviewer: nit: line 23 wording, `make changes on your locally on your computer`

If your change is only a few lines long, or you're not that familiar with git, read [Small changes](#small-changes) to learn how to quickly edit a page.

> If your changes are larger, or span multiple pages, read [Fork the kubernetes/website repostiory](#fork-the-repo) to learn how to fork the Kubernetes website repository.

Reviewer: nit: typo, `repository`

{{% /tab %}}

> ### Open a pull request from your fork to kubernetes/website

Reviewer: nit: line 198, typo, s/subit/submit/
